packages/http/libraries/core/src/http/guard/Guard.ts

```diff
@@ -0,0 +1,36 @@
+import { inject, injectable } from 'inversify';
+
+import { HttpAdapter } from '../adapter/HttpAdapter';
+import { InversifyHttpAdapter } from '../adapter/InversifyHttpAdapter';
+import { Middleware } from '../models/Middleware';
+
+@injectable()
+export abstract class Guard<
+  TRequest,
+  TResponse,
+  TNextFunction extends () => void,
+> implements Middleware<TRequest, TResponse, TNextFunction>
```

Comment:
Why does `Guard` extend `Middleware`? I think `Guard should not receive responses nor next functions. Consider  #475 as reference

Reply:
Here I am considering that a guard is a middleware with more logic. This way, we can treat guards as if they were middlewares with a specific order in the adapter. Anyway, I will try to simplify the guard's contract.